enterprise_access/apps/content_assignments/content_metadata_api.py

```diff
@@ -28,11 +28,28 @@ def get_content_metadata_for_assignments(enterprise_catalog_uuid, assignments):
         to a content metadata dictionary, or null if no such dictionary
         could be found for a given key.
     """
-    content_keys = sorted({assignment.content_key for assignment in assignments})
-    content_metadata_list = get_and_cache_catalog_content_metadata(enterprise_catalog_uuid, content_keys)
-    metadata_by_key = {
-        record['key']: record for record in content_metadata_list
+    content_keys = {
+        (assignment.content_key, assignment.is_assigned_course_run)
+        for assignment in assignments
     }
+    content_metadata_list = get_and_cache_catalog_content_metadata(enterprise_catalog_uuid, content_keys)
+    metadata_by_key = {}
+    for record in content_metadata_list:
+        record_key = record.get('key')
+
+        # Now, check if the record_key matches either the content_key or parent_content_key in the original content_keys
+        for content_key, is_assigned_course_run in content_keys:
+            if is_assigned_course_run:
+                metadata_by_key.update({
+                    content_key: record
+                })
+                break  # Stop searching after the match is found
+            if record_key == content_key:
+                metadata_by_key.update({
+                    record_key: record
+                })
+                break  # Stop searching after the match is found
```

Comment:
There's something weird going on here cause both if-blocks do the same thing so they could just be collapsed into one (like below), but that doesn't seem right because if the first row of `content_keys` corresponds to an assigned course run, then metadata_by_key would always only have 1 key in it.
```suggestion
            if is_assigned_course_run or record_key == content_key:
                metadata_by_key.update({
                    content_key: record
                })
                break  # Stop searching after the match is found
```

I haven't thought through what the solution would be, just flagging it for now.

Reply:
> There's something weird going on here cause both if-blocks do the same thing so they could just be collapsed into one (like below)

I don't believe the if-blocks were doing quite the same thing? The first one dealing with `if is_assigned_course_run` uses the assignment's `content_key` whereas the second one regarding `if record_key == content_key:` uses the top-level course key.

The above suggestion would both rely on the assignment's `content_key`, or the course run key in this case and then fail to lookup the content metadata based on the `assignment.content_key` to top-level courses.

I just pushed up a new commit https://github.com/openedx/enterprise-access/pull/561/commits/69b0ff2ddcc9945b023d09d8d6c426682dbba7ba that might improve the readability/understandability of the logic some?